lib/Conversion/ImportVerilog/ImportVerilogInternals.h

```diff
@@ -0,0 +1,73 @@
+//===- ImportVerilogInternals.h - Internal implementation details ---------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// NOLINTNEXTLINE(llvm-header-guard)
+#ifndef CONVERSION_IMPORTVERILOG_IMPORTVERILOGINTERNALS_H
+#define CONVERSION_IMPORTVERILOG_IMPORTVERILOGINTERNALS_H
+
+#include "circt/Conversion/ImportVerilog.h"
+#include "circt/Dialect/Moore/MooreOps.h"
+#include "slang/ast/ASTVisitor.h"
+#include "llvm/Support/Debug.h"
+#include <map>
+#include <queue>
+
+#define DEBUG_TYPE "import-verilog"
+
+namespace circt {
+namespace ImportVerilog {
+
+/// A helper class to facilitate the conversion from a Slang AST to MLIR
+/// operations. Keeps track of the destination MLIR module, builders, and
+/// various worklists and utilities needed for conversion.
+struct Context {
```

Comment:
Later on we'll have additional conversion methods here, stuff like `convertExpression`, `convertType`, `convertStatement`, and the like. Module body conversion then calls type conversion for ports and statement conversion for some of the body, and statement conversion may call into expression conversion, etc.

The MLIR op structure is likely to not have a 1-to-1 correspondance with the AST. For example, Slang has a tendency to inline the module bodies into instances (which we'll have to fix eventually, there's some upstream work there), which means instances you encounter in the module body may need to create global ops. Similarly statements, which may need to create global ops for function declarations and the like. That's also why there's a module worklist in `Context` right now -- it's not entirely clear all instances are truly limited to the `convertModuleBody` function.

Reply:
Ah, I see. For some reason, I was thinking of a more modular/distributed architecture, but this is not really necessary here and complicates things. I guess the name of this class was part of the reason why I thought of this because, in my mind, a `Context` is a collection of 'almost global' data and functionality that every part of the overall (sub)system has access to, but here it doesn't seem to be passed around to provide additional context. Sorry for all this nitpicking about names, but don't you think something like `SlangToMooreConverter` would make it more obvious what this class is doing?

Reply:
I like `Converter`. That sounds like it may be a bit more clear, and there's less overlap with `MLIRContext`. Let me take a note of that and renamed the thing after a bunch of additional PRs have gone in, to avoid widespread renaming on the `slang-frontend` branch.